lib/engines/hcktest/playlist.rb

```diff
@@ -52,36 +57,61 @@ def ms_playlist(log)
       workspace_file
     end
 
+    def custom_playlist
+      playlist = @project.options.test.playlist
+
+      workspace_file = "#{@project.workspace_path}/playlist_#{@kit[3..]}.xml"
+
+      FileUtils.cp(playlist, workspace_file)
+
+      @logger.info("Applying custom kit playlist from #{playlist}")
+      workspace_file
+    end
+
     def sort_by_duration
       @tests.each_with_index do |x, i|
         @tests[i]['duration'] = time_to_seconds(x['estimatedruntime'])
       end
       @tests.sort_by! { |test| test['duration'] }
     end
 
-    def custom_playlist(log)
-      playlist = @project.engine.target['playlist']
-      return unless playlist
+    def custom_select_test_names(log)
+      user_test_names_file = @project.options.test.select_test_names
+
+      select_test_names = if user_test_names_file.nil?
+                            @project.engine.target['select_test_names']
+                          else
+                            File.readlines(user_test_names_file, chomp: true)
+                          end
+
+      return unless select_test_names
 
-      @tests.select! { |test| playlist.include?(test['name']) }
+      @tests.select! { |test| select_test_names.include?(test['name']) }
       count = @tests.count
-      @logger.info("Applying custom playlist, #{count} tests.") if log
+      @logger.info("Applying custom selected test names, #{count} tests.") if log
     end
 
-    def custom_ignore_list(log)
-      ignore_list = @project.engine.target['ignore_list']
-      @ignored_list = []
-      if ignore_list
-        @tests.reject! do |test|
-          if ignore_list.include?(test['name'])
-            @ignored_list << test
-            true
-          else
-            false
-          end
+    def custom_reject_test_names(log)
+      custom_reject_test_names_file = @project.options.test.reject_test_names
```

Comment:
The word `custom` is overloaded; it intended to denote it's from user, here but the method name also includes the word to describe it's not from XML. Instead, use the word `user` as `custom_select_test_names` does.

```suggestion
      user_reject_test_names_file = @project.options.test.reject_test_names
```